plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -2234,7 +2234,12 @@ private TableStatistics getTableStatistics(String tableName, Constraint constrai
                 .execute(getSession(), session -> {
                     Optional<TableHandle> optionalHandle = metadata.getTableHandle(session, qualifiedName);
                     checkArgument(optionalHandle.isPresent(), "Could not create table handle for table %s", tableName);
-                    return metadata.getTableStatistics(session, optionalHandle.get(), constraint);
+                    TableHandle tableHandle = optionalHandle.get();
+                    Optional<ConstraintApplicationResult<TableHandle>> result = metadata.applyFilter(session, tableHandle, constraint);
+                    if (result.isPresent()) {
+                        tableHandle = result.get().getHandle();
+                    }
+                    return metadata.getTableStatistics(session, tableHandle);
```

Comment:
The `Constraint` passed here was so that we exercise this line

https://github.com/trinodb/trino/blob/a585a57b1d0e60fb996e671e54244113f15f6aba/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsMaker.java#L169

1. the line is no longer needed
2. the tests are no longer needed
3. so, this can be simplified.

Reply:
I removed this whole test. But I haven't checked if there are other tests that would rely on getting table stats with predicates (after calling `applyFilter`).